apps/extension/src/popup.ts

```diff
@@ -18,9 +24,10 @@ const isChromeResponderDroppedError = (
 export const popup = async <M extends PopupMessage>(
   req: PopupRequest<M>,
 ): Promise<M['response']> => {
-  await spawnPopup(req.type);
-  // We have to wait for React to bootup, navigate to the page, and render the components
-  await new Promise(resolve => setTimeout(resolve, 800));
+  const popupId = crypto.randomUUID();
+  await spawnPopup(req.type, popupId);
+  await popupReady(popupId);
```

Comment:
suggestion: we should add a code comment here that informs devs that this is necessary given it takes a bit of time for the popup to be ready to accept messages.